### ISSUE TITLE
Lint/UnneededDisable autocorrections never applied

### ISSUE THREAD

**Issue by segiddins** [bbatsov/rubocop#2941]
Autocorrections for Lint/UnneededDisable are never applied, even though the autocorrections are reported by rubocop.

## Steps to reproduce the problem

```
$ cat f.rb
# encoding: utf-8
# frozen_string_literal: true
module Foo
  # rubocop:disable Style/Alias
  attr_reader :foo
  # rubocop:enable Style/Alias
end

$ rubocop -a f.rb
Inspecting 1 file
W

Offenses:

ruby.rb:4:3: W: [Corrected] Unnecessary disabling of Style/Alias.
  # rubocop:disable Style/Alias
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense corrected

$ rubocop -a f.rb
Inspecting 1 file
W

Offenses:

ruby.rb:4:3: W: [Corrected] Unnecessary disabling of Style/Alias.
  # rubocop:disable Style/Alias
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense corrected
```

## RuboCop version

Include the output of `rubocop -V`: (0b8721ee964da3c2ba8382004f24b73f74b1287d)

```
$ rubocop -V
0.38.0 (using Parser 2.3.0.6, running on ruby 2.3.0 x86_64-darwin15)
```

**Comment by alexdowad** [comment]
I've found the problem. This may have never actually worked since I added it last December.

**Comment by bbatsov** [comment]
:-) Better late than never!

**Comment by alexdowad** [comment]
Sorry I haven't sent a fix in. The problem is that `UnneededDisable` does not participate in the normal inspection/autocorrection loop. It is explicitly called at a later stage. But that later stage doesn't apply any autocorrections which it finds.

**Comment by grosser** [comment]
FYI still seeing this in 0.40
